scalameta/transversers/shared/src/main/scala/scala/meta/transversers/Api.scala

```diff
@@ -2,33 +2,58 @@ package scala.meta
 package transversers
 
 private[meta] trait Api {
+
+  sealed trait TraversalOrder
+  case object LeafFirst extends TraversalOrder
+  case object RootFirst extends TraversalOrder
```

Comment:
How about factoring out these definitions into `TraversalOrder.scala` and then aliasing them here if you feel that these guys belong to the public API? This is the pattern that we're using in other modules of Scalameta.